pwiz_tools/Skyline/Util/PanoramaUtil.cs

```diff
@@ -773,32 +776,83 @@ public CacheFormatVersion GetSupportedSkydVersion(FolderInformation folderInfo)
             return CacheFormatVersion.CURRENT;
         }
 
-        public ShareType DecideShareType(FolderInformation folderInfo, SrmDocument document)
+        public ShareType DecideShareTypeVersion(FolderInformation folderInfo, SrmDocument document, ShareType shareType)
         {
-            ShareType shareType = ShareType.DEFAULT;
-            
-            var settings = document.Settings;
-            Assume.IsTrue(document.IsLoaded);
-            var cacheVersion = settings.HasResults ? settings.MeasuredResults.CacheVersion : null;
+            var cacheVersion = GetDocumentCacheVersion(document);
 
             if (!cacheVersion.HasValue)
             {
                 // The document may not have any chromatogram data.
                 return shareType;
             }
 
-            CacheFormatVersion supportedVersion = GetSupportedSkydVersion(folderInfo);
+            var supportedVersions = GetSupportedVersionsForCacheFormat(folderInfo, cacheVersion);
+            CacheFormatVersion supportedVersion = supportedVersions.Item1;

```

Comment:
Use a custom struct with descriptive field names. There are very few cases where I think a Tuple is an acceptable replacement for named fields, which provide much better in-code documentation. Definitely not as a return type from a public function.

Reply:
I think that function should be changed to just return a SkylineVersion. The SkylineVersion has a "CacheFormatVersion" in it, and I think that's the CacheFormatVersion that should end up getting used.